### PR TITLE
Add a macOS worker to the CI

### PR DESCRIPTION
**Pull request by asmeurer**
## Description

This is built on https://github.com/conda-incubator/conda-store/pull/549 but I've made in a separate PR because I'm not sure if there will be other issues here and I don't want to block that PR on this (but at the same time, tests won't pass on Mac without the changes from that PR).

See #513 and #507

## Pull request checklist

- [ ] Did you test this change locally?
- [ ] Did you update the documentaion (if required)?
- [ ] Did you add/update relevant tests for this change (if required)?

### CODE REVIEW

**Comment by asmeurer** [comment]
The setup-miniconda stage failed with

```
Updating 'conda-store-server-dev' env from conda env update...
  /usr/local/miniconda/condabin/mamba env update --name conda-store-server-dev --file conda-store-server/environment-dev.yaml
  Warning: Traceback (most recent call last):
    File "/usr/local/miniconda/condabin/mamba", line 7, in <module>
  
  Traceback (most recent call last):
    File "/usr/local/miniconda/condabin/mamba", line 7, in <module>
  Warning:     from mamba.mamba import main
    File "/usr/local/miniconda/lib/python3.11/site-packages/mamba/mamba.py", line 49, in <module>
      import libmambapy as api
  
      from mamba.mamba import main
    File "/usr/local/miniconda/lib/python3.11/site-packages/mamba/mamba.py", line 49, in <module>
      import libmambapy as api
  Warning:   File "/usr/local/miniconda/lib/python3.11/site-packages/libmambapy/__init__.py", line 7, in <module>
  
    File "/usr/local/miniconda/lib/python3.11/site-packages/libmambapy/__init__.py", line 7, in <module>
  Warning:     raise e
    File "/usr/local/miniconda/lib/python3.11/site-packages/libmambapy/__init__.py", line 4, in <module>
  
      raise e
    File "/usr/local/miniconda/lib/python3.11/site-packages/libmambapy/__init__.py", line 4, in <module>
  Warning:     from libmambapy.bindings import *  # noqa: F401,F403
  
      from libmambapy.bindings import *  # noqa: F401,F403
  Warning:     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  ImportError: dlopen(/usr/local/miniconda/lib/python3.11/site-packages/libmambapy/bindings.cpython-311-darwin.so, 0x0002): Library not loaded: '@rpath/libarchive.13.dylib'
    Referenced from: '/usr/local/miniconda/lib/libmamba.2.0.0.dylib'
    Reason: tried: '/usr/local/miniconda/lib/libarchive.13.dylib' (no such file), '/usr/local/miniconda/lib/python3.11/site-packages/libmambapy/../../../libarchive.13.dylib' (no such file), '/usr/local/miniconda/lib/python3.11/site-packages/libmambapy/../../../libarchive.13.dylib' (no such file), '/usr/local/miniconda/bin/../lib/libarchive.13.dylib' (no such file), '/usr/local/miniconda/bin/../lib/libarchive.13.dylib' (no such file), '/usr/local/lib/libarchive.13.dylib' (no such file), '/usr/lib/libarchive.13.dylib' (no such file)
  
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  ImportError: dlopen(/usr/local/miniconda/lib/python3.11/site-packages/libmambapy/bindings.cpython-311-darwin.so, 0x0002): Library not loaded: '@rpath/libarchive.13.dylib'
    Referenced from: '/usr/local/miniconda/lib/libmamba.2.0.0.dylib'
    Reason: tried: '/usr/local/miniconda/lib/libarchive.13.dylib' (no such file), '/usr/local/miniconda/lib/python3.11/site-packages/libmambapy/../../../libarchive.13.dylib' (no such file), '/usr/local/miniconda/lib/python3.11/site-packages/libmambapy/../../../libarchive.13.dylib' (no such file), '/usr/local/miniconda/bin/../lib/libarchive.13.dylib' (no such file), '/usr/local/miniconda/bin/../lib/libarchive.13.dylib' (no such file), '/usr/local/lib/libarchive.13.dylib' (no such file), '/usr/lib/libarchive.13.dylib' (no such file)
```

**Comment by costrouc** [comment]
:crossed_fingers: looking forward to this one!

**Comment by trallard** [comment]
@asmeurer is there anything else that needs doing?

**Comment by asmeurer** [comment]
The CI is failing in the initial setup stage of setup-miniconda. I'm not sure why. Since the error occurs in the setup stage, it seems like it would be an issue with setup-miniconda itself. And I haven't changed any part of that in the CI config, all I did was add macos as a platform to the matrix. 

It would help if someone more familiar with that could take a look (like Jaime, or if he is available, Gonzalo since he seems to be the main maintainer of setup-miniconda).

I can't really do much until the CI works, but I expect that things ought to pass once they do, since the tests working for me locally on my Mac.

**Comment by asmeurer** [comment]
This is working now. I believe the test failure is the one I noted on Slack (I'll open an issue later), which is related to the paths being too long.

**Comment by asmeurer** [comment]
Oh actually the one I posted on Slack is different. I get that locally but it doesn't happen on CI. The issue here is I just need to disable Docker from being tested on Mac.

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -26,14 +25,23 @@ jobs:
       - name: "Checkout Repository"
         uses: actions/checkout@v4
 
-      - name: Set up Python
+      - name: Set up Python (Linux)
+        if: matrix.os == 'ubuntu-latest'
         uses: conda-incubator/setup-miniconda@v2
         with:
           mamba-version: "*"
```
Comment (asmeurer) [inline .github/workflows/tests.yaml:32]: @trallard is removing this what you had in mind? It seems to work. The install on CI is a few minutes slower but I'm not sure if it's significant enough to worry about.

**Review by trallard — APPROVED** [review]
This looks good and CI is passing now so will merge

File: .github/workflows/tests.yaml (hunk 1 above)
Comment (asmeurer) [reply .github/workflows/tests.yaml:32]: I think this is doing the opposite, using conda only and not using mamba at all. But like I said, it's not *that* much slower.

**Pull request merged**